### PR TITLE
[#42] [Backend] As a user, I can see the survey questions

### LINKED ISSUE #42: [Backend] As a user, I can see the survey questions
## Why

When a user starts taking a survey, he/she will go through a set of questions of the selected survey. The mobile app should show how many questions there are for the survey so that the user can expect how long it is going to take and he/she can always stop at any point by dismissing the screen.

## Acceptance Criteria

- Call `GET /api/v1/surveys/{{survey-id}}` endpoint to retrieve the survey details
- Map the response with the only necessary values for now

## Resources
[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0)

### PR DESCRIPTION
**Pull request by suho**
- Closed #42

## What happened 👀

I created a backend for getting survey detail: including questions and answers.

## Insight 📝

- I updated the Survey model to have questions and answers.
- I added tests for SurveyRepository and the new use case.

## Proof Of Work 📹

Screenshot 2022-11-25 at 14 50 08

Screenshot 2022-11-25 at 15 57 52

### CODE REVIEW

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Survey.kt [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package co.nimblehq.ic.kmm.suv.domain.model
 
 data class Survey(
+    val id: String,
     val title: String,
     val description: String,
     val isActive: Boolean,
-    val coverImageUrl: String
+    val coverImageUrl: String,
+    val questions: List<Question>? = null
```
Comment (luongvo) [inline shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Survey.kt:9]: Should we have a `sortedQuestions` getter here to get all questions of a survey sorted by question.displayOrder?

**Review by luongvo — APPROVED** [review]
rest lgtm, the `sortedQuestions` getter could be added in integration task 👌

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Survey.kt (hunk 1 above)
Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Survey.kt:9]: Fixed in https://github.com/suho/kmm-ic/pull/113/commits/f771a28f0b85fecfc91063c1c64677166476cd53

**Comment by suho** [comment]
As @hoangnguyen92dn approved [this PR](https://github.com/suho/kmm-ic/pull/110). I will merge this one :D

**Pull request merged**